src/subclass/widget.rs

```diff
@@ -0,0 +1,249 @@
+use gtk_sys;
+
+use glib::translate::*;
+
+use glib::subclass::prelude::*;
+
+use crate::Orientation;
+use Widget;
+use WidgetClass;
+
+pub trait WidgetImpl:
+    WidgetImplExt + ObjectImpl + 'static
+{
+    fn adjust_baseline_allocation(&self, widget: &Widget, baseline: &mut i32) {
+        self.parent_adjust_baseline_allocation(widget, baseline)
+    }
+
+    fn adjust_baseline_request(&self, widget: &Widget, minimum_baseline: &mut i32, natural_baseline: &mut i32) {
+        self.parent_adjust_baseline_request(widget, minimum_baseline, natural_baseline)
+    }
+
+    fn adjust_size_allocation(&self, widget: &Widget, orientation: Orientation, minimum_size: &mut i32, natural_size: &mut i32, allocated_pos: &mut i32, allocated_size: &mut i32) {
+        self.parent_adjust_size_allocation(widget, orientation, minimum_size, natural_size, allocated_pos, allocated_size)
+    }
+
+    fn adjust_size_request(&self, widget: &Widget, orientation: Orientation, minimum_size: &mut i32, natural_size: &mut i32) {
+        self.parent_adjust_size_request(widget, orientation, minimum_size, natural_size)
+    }
+
+    fn button_press_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
+        self.parent_button_press_event(widget, event)
+    }
+
+    fn button_release_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
```

Comment:
The return value here and above should probably be `gtk::Inhibit`

Reply:
Same for `button_press_event`?

Reply:
Yes